internal/status/httproute.go

```diff
@@ -16,29 +13,17 @@ import (
 // Extend support to cover more cases.
 func prepareHTTPRouteStatus(
 	status state.HTTPRouteStatus,
-	gwNsName types.NamespacedName,
 	gatewayCtlrName string,
 	transitionTime metav1.Time,
 ) v1beta1.HTTPRouteStatus {
 	parents := make([]v1beta1.RouteParentStatus, 0, len(status.ParentStatuses))
 
-	// FIXME(pleshakov) Maintain the order from the HTTPRoute resource
```

Comment:
Is the order preserved when we create the `status.ParentStatuses`?

Reply:
yep. 
when building `ParentStatuses`, we iterate over `route.ParentRefs`, which is a slice, with the order same as in the ParentRefs of the HTTPRoute in the Spec.